larq_compute_engine/mlir/python/graphdef_tfl_flatbuffer.cc

```diff
@@ -58,12 +58,23 @@ pybind11::bytes ConvertGraphDefToTFLiteFlatBuffer(
     throw std::runtime_error("Invalid target.");
   }
 
+  // `ParseInputArrayInfo` requires a type that isn't pybind compatible, so
+  // translate here.
+  std::vector<llvm::Optional<std::vector<int>>> translated_input_shapes;
```

Comment:
Won't pybind recognise this type if we change the signature of `ConvertGraphDefToTFLiteFlatBuffer` to accept `std::vector<llvm::Optional<std::vector<int>>>`?

Reply:
I don't think so, no. I tried that previously and got an error. I think that makes sense though because pybind has no way of knowing how to construct an element with type `llvm::Optional`.